POM/dashboardPage.js

```diff
@@ -2,14 +2,19 @@ const BasePage = require('./basePage');
 
 class DashboardPage extends BasePage {
   constructor(page) {
-    super(page);  // Pass the page object to the base class
-    this.addDashboardButton = this.page.locator('.addDashboardButton__add-dashboard-btn--acseh');  // Locator
+    super(page); // Pass the page object to the base class
+    this.addDashboardButton = this.page.locator('.addDashboardButton__add-dashboard-btn--acseh'); // Locator
     this.dashboardName = this.page.locator('.input__input--iYEmM');
     this.dashboardDescription = this.page.locator('.inputTextArea__input-text-area--N0goa');
     this.addButton = this.page.locator('//button[text()="Add"]');
     this.addWidgetButton = this.page.locator('//span[text()="Add new widget"]').nth(1);
     this.nextStepButton = this.page.locator('//span[text()="Next step"]');
     this.noWidgetMessage = this.page.locator('.emptyWidgetGrid__empty-widget-headline--ZocRI');
+    this.demoFilter = this.page.locator('.widgetHeader__widget-name-block--AOAHS');
+    this.addFilterButton = this.page.locator('//span[text()="Add filter"]');
+    this.filterNameInput = this.page.locator('input[placeholder="Input filter name"]');
+    this.launchNameInput = this.page.locator('input[placeholder="Enter name"]');
+  
   };
 
   async addDasboard(dashboardNameText, dashboardDescriptionText) {
```

Comment:
addDasboard -> addDashboard?
you can install spell checker extension if you use vscode